packages/gatsby-cli/src/structured-errors/construct-error.ts

```diff
@@ -8,7 +8,7 @@ interface IConstructError {
   details: {
     id?: ErrorId
     context?: Record<string, string>
-    error?: string
+    error?: Error
```

Comment:
@ascorbic is this correct here? This type was added today in https://github.com/gatsbyjs/gatsby/pull/20597, but I don't think you touched it recently

Reply:
(it causes failing typecheck without this change because
```
sanitizeStructuredStackTrace(stackTrace.parse(otherDetails.error))
```

`StackTrace.parse` accept only Errors and not strings

Reply:
Ok, I'll assume this is correct change because @ascorbic does similar in https://github.com/gatsbyjs/gatsby/pull/22252